README.md

```diff
@@ -56,37 +138,118 @@ cd ./server/server
+
+## Development
+
+### Project Directory structure
+This project is setup to have multiple packages and services in the same repository (mono-repository) as such the 
+directory structure is quite important. Below, we briefly described, maybe not all, but the most important 
+directories of the project.
+
+```
+PLATFORM
+├─ LICENSE
+├─ README.md
+├─ CONTRIBUTING.md
+├─ package.json
+├─ rush.json
+├─ common - contains general configurtations for the project
+│  ├─ changes
+│  ├─ config
+│  ├─ git-hooks
+│  └─ scripts
+├─ dev - contains local development server
+├─ models - contains definitions for domain modals
+├─ packages - contains code shared among client and server, except packages/platform which defines plugin architecture and client-side only. 
+├─ plugins - contains client-side code packaged in form of Platform Plugins.
+├─ products - contains examples of business aplications
+│  └─ hr-build
+├─ server - contains server-side code of the Platform
+└─ tests - contains automated tests for the Platform code
+   └─ server
+```
 
-### Running Dev server with auth/workspace
+ 
+### How to run dev server
 
-It will create workspace 'workspace' users 'john.appleseed@gmail.com' and 'brian.appleseed@gmail.com' with passwords '123'. 
-So login functionality could be used to test with different accounts.
+> Before, please make sure that MongoDB is started and available.
+ 
+It needs to run 2 parts in parallel - server & site.
+
+#### Running server
 
 ```bash
 cd ./dev/server
 rushx dev
 ```
 
-### Running ElasticSearch in docker
+FYI: It will create the `workspace` workspace and users `john.appleseed@gmail.com` and `brian.appleseed@gmail.com` with passwords `123`. 
+So, login functionality could be used to test with different accounts.
+
+
+#### Running site
 
 ```bash
+cd ./dev/prod
+rushx dev-server
+```
+
+That's it! Your application is available at http://localhost:8080
+(site) and http://localhost:18080 (server)
 
-docker run -p 9200:9200 -p 9300:9300 -e "discovery.type=single-node" elasticsearch/elasticsearch:7.13.1
+### How to run only site (in-memory storage)
+
+If you want to work with UI only and storage is not important you can run it the following way
+
+```bash
+cd ./dev/prod
+rushx dev
+```
+
+
+### Dev tasks
+
+All dev required tasks could be executed from root of project using `rushx` tool.
+
+#### Force all parts of the project to be rebuilt
+```bash
+rush rebuild
+```
+
+#### Format all changed files
+
+```bash
+rush format
+```
+
+#### Lint all changed files
+
+```bash
+rush lint
+```
+
+#### Execute tests on modified projects
+
+```bash
+rush test
 ```
 
-## Dev tasks
+## Troubleshooting
+This section covers common issues that developers may encounter when executing this project.
 
-All dev required tasks could be execured from root of project using rushx tool.
+### MacOS on M1
 
-### Format all changed files
+If you have MacOS on M1 it needs to use Node.js 16.x for the project.
```

Comment:
Better to put this in section about node jabs requirements, in this case it will be no troubleshooting required.